### ISSUE TITLE
API for determining if user manually moved the map

### ISSUE THREAD

**Issue by jerryryle** [react-native-maps/react-native-maps#2756]
### Is this a bug report?

No

### Have you read the [Installation Instructions](https://github.com/react-community/react-native-maps/blob/master/docs/installation.md)?

Yes

### Environment

react-native: 0.59.1
react: 16.8.3
react-navigation: 3.3.2
react-native-maps: 0.23.0
iOS+Android

### Feature Request Description

A common use case for maps seems to be to track the user's location ("follow me") until the user manually moves the map--at which point, the app might display a crosshair or similar icon to allow the user to re-enable "follow me" mode. To implement this, we need to know whether a map's region update was the result of a programmatic update or due to user input. Currently, there doesn't seem to be an authoritative means for determining this without impacting the UX. Under iOS, using `onPanDrag` only works if `scrollEnabled` is `false`, so this event isn't useful for this scenario (and I'm not sure if it fires if a user manages to zoom without panning). Using the `onMoveShouldSetResponder` has been proposed as a solution, but it seems to have issues interacting with [react-native-gesture-handler](https://github.com/kmagiera/react-native-gesture-handler), which popular libraries such as [react-navigation](https://github.com/react-navigation/react-navigation) use. Regardless, using the likes of `onMoveShouldSetResponder` seems burdensome and potentially conflicting with the map gesture handlers, so it would be nice if `MapView` exposed an official API for determining whether the user has panned or zoomed the map.

Would it be possible to add something like a `userInputChanged` boolean to the `onRegionChange` and `onRegionChangeComplete` events?

**Comment by oliverdolgener** [comment]
I'd like to do the same thing. Currently I'm using `onPanDrag()` on Android which works exactly the way I want. On iOS though I use `onRegionChangeComplete()` and check if the user's current position is inside the viewport region. If not I disable `follow me`. But a consistent solution for both platform would be awesome!

**Comment by failbottt** [comment]
@oliverdolgener 

> `On iOS though I use onRegionChangeComplete() and check if the user's current position is inside the viewport region. `

Mind sharing how you check if the user's current position is inside of the viewport?

**Comment by oliverdolgener** [comment]
```
isCoordinateInRegion(coordinate, region) {
  const top = region.latitude + region.latitudeDelta / 2;
  const bot = region.latitude - region.latitudeDelta / 2;
  const right = region.longitude + region.logitudeDelta / 2;
  const left = region.longitude - region.logitudeDelta / 2;

  return coordinate.latitude <= top
    && coordinate.latitude >= bottom
    && coordinate.longitude <= right
    && coordinate.longitude >= left;
}
```

**Comment by outaTiME** [comment]
> isCoordinateInRegion(coordinate, region) {
>   const top = region.latitude + region.latitudeDelta / 2;
>   const bot = region.latitude - region.latitudeDelta / 2;
>   const right = region.longitude + region.logitudeDelta / 2;
>   const left = region.longitude - region.logitudeDelta / 2;
> 
>   return coordinate.latitude <= top
>     && coordinate.latitude >= bottom
>     && coordinate.longitude <= right
>     && coordinate.longitude >= left;
> }

Typo errors fixed:

```
isCoordinateInRegion(coordinate, region) {
  const top = region.latitude + region.latitudeDelta / 2;
  const bottom = region.latitude - region.latitudeDelta / 2;
  const right = region.longitude + region.longitudeDelta / 2;
  const left = region.longitude - region.longitudeDelta / 2;

  return coordinate.latitude <= top
    && coordinate.latitude >= bottom
    && coordinate.longitude <= right
    && coordinate.longitude >= left;
}
```

**Comment by yairopro** [comment]
Since `onPanDrag` is triggered on every drag event, it increases the traffic on the bridge making the map's drag animation a little lagging.
I think what it would be great is to have a callback `onPanDragged` which would be called only once the user had finished to drag.

**Comment by jonathan-bibas** [comment]
Help

**Comment by yairopro** [comment]
🐸

**Comment by webraptor** [comment]
Any idea when a npm release will go out?

**Comment by yairopro** [comment]
@webraptor You can still set the github link to the PR's commit with the feature as a dependency in your `package.json`.

**Comment by webraptor** [comment]
@yairopro I know, but still ...

**Comment by outaTiME** [comment]
no stale

**Comment by christopherdro** [comment]
Fixed via 1f47e5bbcb7b24f21e11108e73af19e17eaf84ff
